editor/src/main/java/io/github/rosemoe/sora/text/ContentReference.java

```diff
@@ -151,8 +151,9 @@ public int read(char[] chars, int offset, int length) {
                 column += toRead;
                 read += toRead;
                 if (read < length && columnCount == column) {
-                    chars[offset + read] = '\n';
-                    read++;
+                    var separator = content.getLine(line).getLineSeparatorChars();
+                    System.arraycopy(separator, 0, chars, offset + read, separator.length);
+                    read += separator.length;
```

Comment:
This may cause the reader read over `length` characters in some conditions. You may consider the `columnCount` = `line.getColumnCount() + line.getLineSeparator() == LineSeparator.CRLF ? 1 : 0`